src/lib.rs

```diff
@@ -53,101 +53,153 @@ use indexmap::IndexMap;
     let box_type;
-    if box_str.ends_with(".box") {
-        box_type = BOX_1_TYPE
-    } else if box_str.ends_with(".box2") {
-        box_type = BOX_2_TYPE
+    if boxed_str.ends_with(".box") {
+        box_type = BOX1_TF
+    } else if boxed_str.ends_with(".box2") {
+        box_type = BOX2_TF
     } else {
-        return Err(anyhow!("The box type is unknown"));
+        return Err(anyhow!("Unknown boxed string: {}", boxed_str));
     };
 
     Ok(box_type.to_vec())
 }
 
-/// Take a feed identity (key) as a string and return the encoded bytes representing the feed type.
-pub fn get_feed_type(feed: &str) -> Result<Vec<u8>> {
+/// Take a feed ID (key) as a string and return the encoded bytes representing the feed type-format.
+pub fn get_feed_type(feed_id: &str) -> Result<Vec<u8>> {
     let feed_type;
-    if feed.ends_with(".ed25519") {
-        feed_type = CLASSIC_FEED_TYPE
-    } else if feed.ends_with(".bbfeed-v1") {
-        feed_type = BB_FEED_TYPE
-    } else if feed.ends_with(".ggfeed-v1") {
-        feed_type = GG_FEED_TYPE
+    if feed_id.ends_with(".ed25519") {
+        feed_type = CLASSIC_FEED_TF
+    } else if feed_id.ends_with(".bbfeed-v1") {
+        feed_type = BENDYBT_FEED_TF
+    } else if feed_id.ends_with(".ggfeed-v1") {
+        feed_type = GABBYGR_FEED_TF
     } else {
-        return Err(anyhow!("The feed type is of an unknown format"));
+        return Err(anyhow!("Unknown feed format: {}", feed_id));
     };
 
     Ok(feed_type.to_vec())
 }
 
-/// Take a message key as a string and return the encoded bytes representing the message type.
-pub fn get_msg_type(msg: &str) -> Result<Vec<u8>> {
+/// Take a message ID as a string and return the encoded bytes representing the message type-format.
+pub fn get_msg_type(msg_id: &str) -> Result<Vec<u8>> {
     let msg_type;
-    if msg.ends_with(".sha256") {
-        msg_type = CLASSIC_MSG_TYPE
-    } else if msg.ends_with(".bbmsg-v1") {
-        msg_type = BB_MSG_TYPE
-    } else if msg.ends_with(".ggmsg-v1") {
-        msg_type = GG_MSG_TYPE
+    if msg_id.ends_with(".sha256") {
+        msg_type = CLASSIC_MSG_TF
+    } else if msg_id.ends_with(".bbmsg-v1") {
+        msg_type = BENDYBT_MSG_TF
+    } else if msg_id.ends_with(".ggmsg-v1") {
+        msg_type = GABBYGR_MSG_TF
     } else {
-        return Err(anyhow!("The message type is unknown"));
+        return Err(anyhow!("Unknown message ID: {}", msg_id));
     };
 
     Ok(msg_type.to_vec())
 }
 
+/// Take a blob ID as a string and return the encoded bytes as a vector.
+pub fn encode_blob(blob_id: &str) -> Result<Vec<u8>> {
+    let mut encoded_blob = get_blob_type(blob_id)?;
+    let dot_index = blob_id
+        .rfind('.')
+        .with_context(|| format!("Invalid blob ID; no dot index was found: {}", blob_id))?;
+    // decode blob substring from base64 str to bytes and append to encoded_blob bytes
+    base64::decode_config_buf(&blob_id[1..dot_index], base64::STANDARD, &mut encoded_blob)?;
```

Comment:
How about we do the same `let base64_str` pattern here too? :)

Reply:
Good call, I'm on it ;)